code/prime-number/ViewModel/src/main/java/ru/unn/agile/PrimeNumber/viewmodel/ViewModel.java

```diff
@@ -0,0 +1,144 @@
+package ru.unn.agile.PrimeNumber.viewmodel;
+
+import javafx.beans.binding.BooleanBinding;
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.beans.property.SimpleStringProperty;
+import javafx.beans.property.StringProperty;
+import javafx.beans.value.ChangeListener;
+import javafx.beans.value.ObservableValue;
+import ru.unn.agile.primenumber.model.PrimeNumberFinder;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringJoiner;
+
+public class ViewModel {
+    private final StringProperty startElement = new SimpleStringProperty();
+    private final StringProperty endElement = new SimpleStringProperty();
+    private final StringProperty outputField = new SimpleStringProperty();
+
+    public BooleanProperty calculationDisabledProperty() {
+        return calculationDisabled;
+    }
+
+    private final BooleanProperty calculationDisabled = new SimpleBooleanProperty();
+    private final StringProperty status = new SimpleStringProperty();
+    private final List<ValueChangeListener> valueChangedListeners = new ArrayList<>();
+
+    public StringProperty startElemProperty() {
+        return startElement;
+    }
+
+    public StringProperty endElemProperty() {
+        return endElement;
+    }
+
+    public StringProperty outputProperty() {
+        return outputField;
+    }
+
+    public StringProperty statusProperty() {
+        return status;
+    }
+
+    public void findPrimaryNums() {
+        if (calculationDisabled.get()) {
+            return;
+        }
+
+        try {
+            PrimeNumberFinder setOfPrimeNums = new PrimeNumberFinder(
+                    Integer.parseInt(startElement.get()), Integer.parseInt(endElement.get()));
+            List<Integer> primeNumsList = setOfPrimeNums.findNumbers();
+            StringJoiner stringJoiner = new StringJoiner(",");
+            for (Integer elem : primeNumsList) {
+                stringJoiner.add(String.valueOf(elem));
+            }
+            outputField.set(stringJoiner.toString());
+            status.set(Status.SUCCESS.toString());
+        } catch (IllegalArgumentException e) {
+            outputField.set(e.getMessage());
+        }
+    }
+
+    public ViewModel() {
+        startElement.set("");
+        endElement.set("");
+        outputField.set("");
+        status.set(Status.WAITING.toString());
+
+        BooleanBinding couldCalculate = new BooleanBinding() {
+            {
+                super.bind(startElement, endElement);
+            }
+
+            @Override
+            protected boolean computeValue() {
+                return getInputStatus() == Status.READY;
+            }
+        };
+        calculationDisabled.bind(couldCalculate.not());
+
+        // Add listeners to the input text fields
+        final List<StringProperty> fields = new ArrayList<>() {
```

Comment:
It will be better to make it global private field and use it instead of startElement, endElement